File: solutions/go/0876-middle-of-the-linked-list.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func middleNode(head *ListNode) *ListNode {
	middle := head
	head = head.Next

	for i := 0; head != nil; i, head = i+1, head.Next {
		if i%2 == 0 {
			middle = middle.Next
		}
	}

	return middle
}

File: solutions/php/0876-middle-of-the-linked-list.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{
    /**
     * @param ListNode $head
     * @return ListNode
     */
    public function middleNode(ListNode $head): ListNode
    {
        $counter = 0;
        $middle = $head;

        while ($head = $head->next) {
            if ($counter++ % 2 === 0) {
                $middle = $middle->next;
            }
        }

        return $middle;
    }
}
